src/router.js

```diff
@@ -189,43 +178,46 @@ router.get('/registrations/:id/return/:returnId', async (request, response) => {
     // Try to parse the incoming ID to make sure it's really a number.
     const existingId = Number(request.params.id);
     if (isNaN(existingId)) {
-      response.status(404).send({message: `Registration ${request.params.id} not valid.`});
-      return;
+      return response.status(404).send({message: `Registration ${request.params.id} not valid.`});
     }
 
     // Check if there's a registration allocated at the specified ID.
     const existingReg = await Registration.findOne(existingId);
     if (existingReg === undefined || existingReg === null) {
-      response.status(404).send({message: `Registration ${existingId} not allocated.`});
-      return;
+      return response.status(404).send({message: `Registration ${existingId} not allocated.`});
     }
 
     // Try to parse the incoming ID to make sure it's really a number.
     const existingReturnId = Number(request.params.returnId);
     if (isNaN(existingReturnId)) {
-      response.status(404).send({message: `Return ${request.params.returnId} not valid.`});
-      return;
+      return response.status(404).send({message: `Return ${request.params.returnId} not valid.`});
     }
 
-    // Check if there's return allocated at the specified ID.
+    // Check if there's a return allocated at the specified ID.
     const existingReturn = await Return.findOne(existingReturnId);
     if (existingReturn === undefined || existingReturn === null) {
-      response.status(404).send({message: `Return ${existingReturnId} not allocated.`});
-      return;
+      return response.status(404).send({message: `Return ${existingReturnId} not allocated.`});
+    }
+
+    // Check if the return is allocated to the specified registration.
+    if (existingReturn.RegistrationId !== request.params.id) {
```

Comment:
Is this correct?

Earlier we said 
```js
    const existingId = Number(request.params.id);
```
and
```js
    const existingReturnId = Number(request.params.returnId);
```

What type is `existingReturn.RegistrationId`? If it's a number then this check will always fail, as you're checking against the incoming PathParam which will be string-ish and not the parsed number we have already converted.